GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -849,9 +852,28 @@ public void PerformRefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> get
                 return;
             }
 
+            // Revision info is read in two parallel steps:
+            // 1. Read current commit, refs, prepare grid etc.
+            // 2. Read all revisions with git-log.
+            // Both these operations can take a long time and is done async.
+            // Step 2. requires that most of the information in 1. is prepared when updating the grid.
+            // Git will will provide this information when available (so slower to first info if sorting).
+            // This information from step 1. is therefore protected with Lazy accessosors and a semaphore.
+            System.Threading.SemaphoreSlim semaphoreCurrentCommit = new(1, 1);
+            semaphoreCurrentCommit.Wait();
             _isRefreshingRevisions = true;
-            ShowLoading();
 
+            GitModule capturedModule = Module;
```

Comment:
❔ Should `capturedModule` be used in the following lines, too, shouldn't it?

Reply:
Yes, does not make a difference there. Module was used in the async code before...

Reply:
Moved this to just before where it is needed, before RunAsync